static/app/views/settings/components/forms/choiceMapperField.tsx

```diff
@@ -140,8 +140,9 @@ export default class ChoiceMapper extends React.Component<FieldProps> {
 
     const removeRow = (itemKey: string) => {
       // eslint-disable-next-line no-unused-vars
-      const {[itemKey]: _, ...updatedValue} = value;
-      saveChanges(updatedValue);
+      saveChanges(
+        Object.fromEntries(Object.entries(value).filter(([key, _]) => key !== itemKey))
```

Comment:
`Object.fromEntries(Object.entries(value).filter(([key]) => key !== itemKey))`

You don't need the underscore, you would only need that if you didn't care about earlier indexes in the array.